src/View/mainpage/IsodoseTab.py

```diff
@@ -1,189 +1,160 @@
-		roiColor[30] = QtGui.QColor(33, 0, 255)
-		roiColor[10] = QtGui.QColor(11, 0, 134)
-		return roiColor
-
-
-	def draw_color_square(self, red, green, blue):
-		"""
-		Create a color square.
-		:param red: Red component of the color.
-		:param green: Green component of the color.
-		:param blue: Blue component of the color.
-
-		:return: Color square widget.
-		"""
-		color_square_label = QtWidgets.QLabel()
-		color_square_pix = QtGui.QPixmap(15, 15)
-		color_square_pix.fill(QtGui.QColor(red, green, blue))
-		color_square_label.setPixmap(color_square_pix)
-		return color_square_label
\ No newline at end of file
+from PyQt5 import QtWidgets, QtGui, QtCore
+
+from src.Model.PatientDictContainer import PatientDictContainer
+
+
+class IsodoseTab(QtWidgets.QWidget):
+
+    request_update_isodoses = QtCore.pyqtSignal()
+
+    def __init__(self):
+        QtWidgets.QWidget.__init__(self)
+        self.patient_dict_container = PatientDictContainer()
+        self.rxdose = self.patient_dict_container.get("rxdose")
+        self.color_dict = self.init_color_isod()
+        self.color_squares = self.init_color_squares()
+        self.checkboxes = self.init_checkboxes()
+
+        self.isodose_tab_layout = QtWidgets.QVBoxLayout()
+        self.isodose_tab_layout.setAlignment(QtCore.Qt.AlignTop)
+        self.isodose_tab_layout.setSpacing(0)
+        self.init_layout()
+
+        self.setLayout(self.isodose_tab_layout)
+
+    def init_layout(self):
+        for i in range(0, len(self.checkboxes)):
+            widget_isodose = QtWidgets.QWidget()
+            layout_isodose = QtWidgets.QHBoxLayout(widget_isodose)
+            layout_isodose.setAlignment(QtCore.Qt.AlignLeft)
+            layout_isodose.addWidget(self.color_squares[i])
+            layout_isodose.addWidget(self.checkboxes[i])
+            self.isodose_tab_layout.addWidget(widget_isodose)
+
+    def init_color_isod(self):
+        """
+        Create a dictionary containing the colors for each isodose.
+
+        :return: Dictionary where the key is the percentage of isodose and the value a QColor object.
+        """
+        roi_color = dict()
+        roi_color[107] = QtGui.QColor(131, 0, 0)
+        roi_color[105] = QtGui.QColor(185, 0, 0)
+        roi_color[100] = QtGui.QColor(255, 46, 0)
+        roi_color[95] = QtGui.QColor(255, 161, 0)
+        roi_color[90] = QtGui.QColor(253, 255, 0)
+        roi_color[80] = QtGui.QColor(0, 255, 0)
+        roi_color[70] = QtGui.QColor(0, 143, 0)
+        roi_color[60] = QtGui.QColor(0, 255, 255)
+        roi_color[30] = QtGui.QColor(33, 0, 255)
+        roi_color[10] = QtGui.QColor(11, 0, 134)
+
+        return roi_color
+
+    def init_color_squares(self):
+        list_of_squares = []
+        for key, value in self.color_dict.items():
+            list_of_squares.append(self.draw_color_square(value))
+
+        return list_of_squares
+
+    def init_checkboxes(self):
+        list_of_checkboxes = []
+        # Values of Isodoses
```

Comment:
this reeks of "should be refactored" because of the DRY ("Don't Repeat Yourself") principle.